xonsh/history.py

```diff
@@ -1,17 +1,22 @@
 # -*- coding: utf-8 -*-
 """Implements the xonsh history object."""
 import argparse
-import functools
 import os
+import sys
+from functools import lru_cache, partial
+from os import listdir
+from operator import itemgetter
 import uuid
 import time
+from datetime import datetime
```

Comment:
Also, since #1220 we merged this morning, these non-package imports on the above lines should be changed to direct imports. See https://github.com/xonsh/xonsh/blob/master/CONTRIBUTING.rst#imports  If you don't want to take care of it, let me know and I will happily do this.  I am going to go ahead and comment what the imports should be, though.

This one should just be `import datetime`